### ISSUE TITLE
Redundant newlines

### ISSUE THREAD

**Issue by jcubic** [XaminProject/handlebars.php#154]
I have newline issue, I have header helpers that create http headers.
```php
        $handlebars->addHelper('header', function($template, $context, $arg, $source) {
            $para = $template->parseArguments($arg);
            header($para[0]);
        });
```

If I have template like this:
```
{{header "Content-Type: text/plain"}}
{{#each foo}}
    {{this}}
{{/sql}}
```
I have redundant newline at the beginning and when I'm using this one:

```
{{header "Content-Type: text/plain"}}{{#each foo}}
    {{this}}
{{/sql}}
```
I have newline at the begining of each iteration (in fact I have two newlines, one expected).

**Comment by JustBlackBird** [comment]
There is a special syntax `{{~smth~}}` for whitespace removing.

Look at [tests](https://github.com/XaminProject/handlebars.php/blob/master/tests/Xamin/HandlebarsTest.php#L331) for details.

**Comment by jcubic** [comment]
This don't remove newlines, If I have first template I don't have any newlines and spaces between items in `#each` and if I use second one I have expected newlines at the end of the items in `#each` but in both cases I have newline at the beginning of the template.

**Comment by jcubic** [comment]
Also if I use
```
{{~header "Content-Type: text/plain"~}}
```
I don't get newline at the begining but I also don't get spaces before `{{this}}`.

**Comment by JustBlackBird** [comment]
Please, add a template you use, the data your template is running on, the expected and the real result. In this case we could talk about your use case in more details.

Be aware, that nobody (except you) has neither `header` nor `sql` helpers, so you should not use them in a provided test cases.

**Comment by jcubic** [comment]
I have code like this:
```
$handlebars->addHelper('bar', function($template, $context, $arg, $source) {
});
$template = '{{bar "baz"}}
{{#each foo}}
    {{this}}
{{/each}}';
echo $handlebars->render($template, array(
    'foo' => array(
        '1', '2', '3', '4'
)));
```
the result is:
```

    1
    2
    3
    4
```
extra newline at the beginning,

expected result:
```
    1
    2
    3
    4
```
no extra newline at the beginning


it don't happen if I use bar helper as block (`#bar`).

**Comment by jcubic** [comment]
Same heppen if you use this template:
```
{{if "baz"}}
{{#each foo}}
    {{this}}
{{/each}}
```

**Comment by JustBlackBird** [comment]
Instead of 

```handlebars
{{bar "baz"}}
{{#each foo}}
    {{this}}
{{/each}}
```
try to use
```handlebars
{{bar "baz"}}
{{~#each foo}}
    {{this}}
{{/each}}
```
(notice the tilde position)

**Comment by jcubic** [comment]
yes it works, thanks. but maybe this can be changed in the library to not introduce new line if empty non block helper is used, same as block helper.

**Comment by JustBlackBird** [comment]
You're welcome.

You see, Handlebars.php tries to work in the same way as Handlebars.js. As I know, JS implementation introduces new line for empty non block helpers. So PHP implementation should follow this behavior. There is little we can do here.

**Comment by jcubic** [comment]
Another issue, this template:
```
{{~bar "baz"~}}
-----------------
{{~#each foo}}
    {{this}}
{{/each}}
```
there is newline before `-----------------`

**Comment by jcubic** [comment]
I have another problem with custom helpers I have template like this:
```
{{#each foo}}
    {{#each bar}}
        {{~../id}}: {{content}}
        {{~helper~}}
        {{#if $id == 2}}
            id = 2
        {{/if}}
    {{/each}}
{{/each}}
```
and I end up with this result:
```

1: lorem
        
2: ipsum
        
            id = 2
3: dolor
        
4: sit
        
5: amet
        
```
and if I add `{{content~}}` I ended up with this:
```
1: lorem2: ipsumid = 2
3: dolor4: sit5: amet
```
